src/main/java/at/ac/fhcampuswien/fhmdb/database/WatchlistRepository.java

```
@@ -9,13 +9,24 @@
 public class WatchlistRepository {
 
     private final Dao<WatchlistMovieEntity, Long> dao;
-
-    public WatchlistRepository() throws DatabaseException
+    private static WatchlistRepository instance;
+    private WatchlistRepository() throws DatabaseException
     {
         Database.getInstance().createTables();
         this.dao = Database.getInstance().getWatchlistDao();
     }
 
+    public static WatchlistRepository getInstance() throws DatabaseException {
+        try {
+            if(instance == null) {
+                instance = new WatchlistRepository();
+            }
+            return instance;
+        } catch (SQLException e) {
```

Why is there a catch block here? We can just propagate the exception without catching (`throws DatabaseException`). Like in the constructor (we don't have a catch there).